server/swap/swap.go

```diff
@@ -1221,48 +1240,39 @@ func (s *Swapper) checkInaction(assetID uint32) {
 
 		switch match.Status {
 		case order.NewlyMatched:
-			if match.makerStatus.swapAsset != assetID {
-				return
-			}
-			// If the maker is not acting, the swapTime won't be set. Check against
-			// the time the match notification was sent (match.time) for the broadcast
-			// timeout.
-			if match.makerStatus.swapTime.IsZero() && match.time.Before(oldestAllowed) {
+			// Maker has not broadcast their swap. They have until match time
+			// plus bTimeout.
+			if match.time.Before(oldestAllowed) {
 				failMatch(true)
 			}
 		case order.MakerSwapCast:
-			if match.takerStatus.swapAsset != assetID {
-				return
-			}
-			// If the maker has sent their swap tx, check the taker's broadcast
-			// timeout against the time of the swap's SwapConf'th confirmation.
-			if match.takerStatus.swapTime.IsZero() &&
-				!match.makerStatus.swapConfirmed.IsZero() &&
+			// If the maker's swap confirmed time (reached swapConf) is more
+			// than bTimeout ago, the match should be in TakerSwapCast.
+			log.Debugf("   MakerSwapCast: taker swap time %v (zero=%v), maker swap confirmed %v (zero=%v), %v ago",
+				match.takerStatus.swapTime, match.takerStatus.swapTime.IsZero(),
+				match.makerStatus.swapConfirmed, match.makerStatus.swapConfirmed.IsZero(), time.Since(match.makerStatus.swapConfirmed))
+			if !match.makerStatus.swapConfirmed.IsZero() &&
 				match.makerStatus.swapConfirmed.Before(oldestAllowed) {
 				failMatch(false)
 			}
 		case order.TakerSwapCast:
-			if match.takerStatus.swapAsset != assetID {
-				return
-			}
-			// If the taker has sent their swap tx, check the maker's broadcast
-			// timeout (for redemption) against the time of the swap's SwapConf'th
-			// confirmation.
-			if match.makerStatus.redeemTime.IsZero() &&
-				!match.takerStatus.swapConfirmed.IsZero() &&
+			// If the taker's swap confirmed time (reached swapConf) is more
+			// than bTimeout ago, the maker should have redeemed.
+			log.Debugf("   TakerSwapCast: maker redeem time %v (zero=%v), taker swap confirmed %v (zero=%v), %v ago",
+				match.makerStatus.redeemTime, match.makerStatus.redeemTime.IsZero(),
+				match.takerStatus.swapConfirmed, match.takerStatus.swapConfirmed.IsZero(), time.Since(match.takerStatus.swapConfirmed))
+			if !match.takerStatus.swapConfirmed.IsZero() &&
 				match.takerStatus.swapConfirmed.Before(oldestAllowed) {
 				failMatch(true)
 			}
 		case order.MakerRedeemed:
-			if match.takerStatus.swapAsset != assetID {
-				return
-			}
 			// If the maker has redeemed, the taker can redeem immediately, so
 			// check the timeout against the time the Swapper received the
 			// maker's `redeem` request (and sent the taker's 'redemption').
-			if match.takerStatus.redeemTime.IsZero() &&
-				!match.makerStatus.redeemTime.IsZero() &&
```

Comment:
This goes further and eliminates the `!match.makerStatus.redeemTime.IsZero()` because this is a bcast time that is set when it becomes `MakerRedeem`, unlike the swap cast states where `swapConfirmed` is only set after SwapConf